test/SILOptimizer/sil_combine_propagate_cta.swift

```diff
@@ -0,0 +1,139 @@
+// RUN: %target-swift-frontend %s -O -wmo -emit-sil -Xllvm -sil-disable-pass=DeadFunctionElimination | %FileCheck %s
+
+internal protocol SomeProtocol : class {
```

Comment:
Please test:

- structs
- public protocol (make sure its not optimized)
- class conforming to non-class-bound protocol
- generic class/struct/enum cannot be optimized
- also check this case:

```
struct Outer<T> {
  struct Inner : MyProto { ... }
}
```

Reply:
@slavapestov I have added tests for all of the above except two. Can you help? 
(1) Can you remind me what you meant by structs? I already have struct test.
(2) For this case:
struct Outer<T> {
  struct Inner : MyProto { ... }
}
The M.Types.getLoweredType call crashes with "Bad base type". Do you have any recommendations on how to obtain lowered SILType for generic nested types like above? I am thinking this case can be handled..

Reply:
FYI, I can also skip the optimization altogether by checking if the concrete type is an unbound generic type using "hasUnboundGenericType()".